x/emissions/types/reputer_value_bundle.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (bundle *ReputerValueBundle) Validate() error {
```

Comment:
This validation could also include the signature check validation that is now currently done when processing the bundle.

Reply:
+1

Reply:
It can be added in the `WorkerDataBundle` validation as well.

Reply:
+ I would also move the Msg Size validations into `Validate()` or `ValidateTopLevel()`, considering the goal of these validations is to perform stateless checks.

Reply:
Think you're both right ab this and all other comments.

Reply:
the size validations are dependent on state, namely a parameter in the keeper so have to keep that